docs/import/import_data.js

```diff
@@ -140,132 +135,327 @@ function stringifyGeoid(feature) {
+  saveBounds(makeGeoJsonRectangle(sw, ne)).catch(firestoreError);
+  return {
+    damage: null,
+    damageEnvelope: damageEnvelope,
+  };
+}
 
-  const snap = ee.FeatureCollection(resources.rawSnap)
-                   .map((feature) => stringifyGeoid(feature, censusGeoidKey));
-  // filter block groups to those with damage.
-  const blockGroups = ee.Join.simple().apply(
-      ee.FeatureCollection(resources.bg), damage,
-      ee.Filter.intersects({leftField: '.geo', rightField: '.geo'}));
+/**
+ * Creates a GeoJson-style rectangle from the southwest and northeast corners.
+ * @param {{lat: number, lng: number}} sw
+ * @param {{lat: number, lng: number}} ne
+ * @return {Object} GeoJson Polygon
+ */
+function makeGeoJsonRectangle(sw, ne) {
+  return {
+    type: 'Polygon',
+    coordinates: [[
+      [sw.lng, sw.lat],
+      [ne.lng, sw.lat],
+      [ne.lng, ne.lat],
+      [sw.lng, ne.lat],
+      [sw.lng, sw.lat],
+    ]],
+  };
+}
 
-  // join snap stats to block group geometries
-  const joinedSnap =
-      ee.Join.inner()
-          .apply(
-              snap, blockGroups,
-              ee.Filter.equals(
-                  {leftField: censusGeoidKey, rightField: tigerGeoidKey}))
-          .map(combineWithSnap);
-  // join with income
-  const joinedSnapIncome =
-      ee.Join.inner()
-          .apply(
-              joinedSnap, ee.FeatureCollection(resources.income),
-              ee.Filter.equals(
-                  {leftField: geoidTag, rightField: censusGeoidKey}))
-          .map(combineWithIncome);
-  // filter SVI to those with damage and join
-  const svi = ee.Join.simple().apply(
-      ee.FeatureCollection(resources.svi), damage,
-      ee.Filter.intersects({leftField: '.geo', rightField: '.geo'}));
-  const joinedSnapIncomeSVI =
-      ee.Join.inner()
-          .apply(
-              joinedSnapIncome.map(addTractInfo), svi,
-              ee.Filter.equals({leftField: tractTag, rightField: cdcGeoidKey}))
-          .map(combineWithSvi);
-  // attach block groups to buildings and aggregate to get block group building
-  // counts
-  const buildings = ee.FeatureCollection(resources.buildings);
+/**
+ * Attaches block groups to buildings and aggregates to get per-block group
+ * building counts.
+ *
+ * Joins the buildings to block groups using a "saveFirst" join, since each
+ * building should be in only one block group, then constructs a histogram based
+ * on the appended block group features.
+ *
+ * This method will go away or be greatly changed if we're using CrowdAI data
+ * instead of previously computed building data.
+ * @param {ee.Geometry.Polygon} damageEnvelope Area we are concerned with
+ * @param {string} buildingPath location of buildings asset in EE
+ * @param {string} state
+ * @param {ee.FeatureCollection} stateGroups Collection with block groups
+ * @return {ee.Dictionary} Number of buildings per block group
+ */
+function computeBuildingsHisto(
+    damageEnvelope, buildingPath, state, stateGroups) {
```

Comment:
unused param state? Don't know why intellij isn't upset about this.

Reply:
Thanks. Yeah, I've noticed that before about intellij with javascript and I don't understand it.